### PR TITLE
lock_mgr: fix busy poll on grpc error in deadlock detector

### PR DESCRIPTION
**Pull request by sticnarf**
### What problem does this PR solve?

The deadlock detector thread can use 100% cpu.

### What is changed and how it works?

 I haven't reproduced this problem but looking at the code I think it's due to the incorrect use of `filter_map`.

When the grpc stream disconnects with error, it always yields an error. In the previous code, the `filter_map` closure thansforms the error to a `None` but the `None` will be filtered by `filter_map` and the stream won't be terminated. So the stream will busy poll infinitely on grpc error.

### Check List 

Tests 

- Manual test

Run a workload that has conflicts. Then, kill a node which is not the leader of the deadlock detector.

With latest master, the cpu of the deadlock detector rises to 100%:

(172.16.4.28 is the deadlock detector leader)

image

With this PR:

(172.16.4.135 is the deadlock detector leader)

image

And now if it disconnects, there is a log now:

```
[2020/09/24 04:58:23.321 +00:00] [WARN] [deadlock.rs:804] ["deadlock detect rpc stream disconnected"] [error=Grpc(RpcFinished(None))]
```

### Release note 

Fix busy polling on grpc error in the deadlock detector.

### CODE REVIEW

**Review by hunterlxt — APPROVED** [review]
LGTM

File: src/server/lock_manager/deadlock.rs [hunk 1]
```diff
@@ -757,55 +757,51 @@ where
...
-                            None
-                        }
-                    }
-                    DeadlockRequestType::CleanUpWaitFor => {
-                        detect_table.clean_up_wait_for(txn.into(), wait_for_txn.into(), *key_hash);
-                        None
-                    }
-                    DeadlockRequestType::CleanUp => {
-                        detect_table.clean_up(txn.into());
+        let mut s = stream.map_err(Error::Grpc).try_filter_map(move |mut req| {
```
Comment (youjiali1995) [inline src/server/lock_manager/deadlock.rs:760]: I remembered there should be a `and_then` before the `filter_map`.. It's removed by https://github.com/tikv/tikv/pull/8597/files#diff-92a18657172bd93f2d8b17a1e93cdd5aR765.

**Review by youjiali1995 — APPROVED** [review]

**Pull request merged**